src/libraries/sfx.proj

```diff
@@ -69,6 +69,27 @@
           UseHardlinksIfPossible="true" />
   </Target>
 
+  <Target Name="ValidateSharedFramework"
+          AfterTargets="Build">
+    <ItemGroup>
+      <!-- exclude private assemblies from ref-->
+      <_expectedRef Include="@(NetCoreAppLibrary)" Condition="!$([System.String]::new('%(Identity)').StartsWith('System.Private'))" />
+      <_expectedSharedFrameworkFile Include="@(_expectedRefs->'$(MicrosoftNetCoreAppRefPackRefDir)%(Identity).dll')" />
+
+      <!-- exclude the full facades from expected docs since they have no types -->
+      <_expectedDoc Include="@(_expectedRefs)" Exclude="netstandard;@(NetFxReference)" />
+      <_expectedSharedFrameworkFile Include="@(_expectedDoc->'$(MicrosoftNetCoreAppRefPackRefDir)%(Identity).xml')" />
+
+      <!-- exclude CoreLib from expected libs, since it's placed in native -->
+      <_expectedLib Include="@(NetCoreAppLibrary)" Exclude="System.Private.CoreLib" />
+      <_expectedSharedFrameworkFile Include="@(_expectedLib->'$(MicrosoftNetCoreAppRuntimePackRidLibTfmDir)%(Identity).dll')" />
+
+      <_missingSharedFrameworkFile Include="@(_expectedSharedFrameworkFile)" Exclude="@(_expectedSharedFrameworkFile->Exists())" />
+    </ItemGroup>
+
+    <Error Text="The shared framework files '@(_missingSharedFrameworkFile)' were missing." Condition="'@(_missingSharedFrameworkFile)' != ''" />
+  </Target>
```

Comment:
I am a big fan of the addition of tests for infra stuff, thank you.

Question: Do you know why the re are some `_missingSharedFrameworkFile` items with the name `''`?

Reply:
> Question: Do you know why the re are some _missingSharedFrameworkFile items with the name ''?

The condition `'@(_missingSharedFrameworkFile)' != ''` means _the item named _missingSharedFrameworkFile is not empty._  It's used to ensure we don't raise the error when no files are missing.